roles/kustomize_deploy/tasks/execute_step.yml

```diff
@@ -40,7 +38,7 @@
         quiet: true
       ignore_errors: true  # noqa: ignore-errors
       register: _builtin_wait_cond_result
-      loop: "{{ stage.wait_conditions }}"
+      loop: "{{ stage.wait_conditions | default([]) }}"
```

Comment:
Would this execute faster if a condition was used on the block? 
Instead of running each task with an empty iteration, all the tasks in the block is skipped?